packages/aws-cdk-lib/aws-s3/lib/notifications-resource/notifications-resource-handler.ts

```diff
@@ -63,14 +63,20 @@ export class NotificationsResourceHandler extends Construct {
       assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com'),
     });
 
-    this.role.addManagedPolicy(
-      iam.ManagedPolicy.fromAwsManagedPolicyName('service-role/AWSLambdaBasicExecutionRole'),
-    );
```

Comment:
Why are we changing this file?

Reply:
because `ManagedPolicy.addManagedPolicy` is being used in that file also, till now which is also **doing Nothing.**
```
      this.role.addManagedPolicy(
        iam.ManagedPolicy.fromAwsManagedPolicyName('service-role/AWSLambdaBasicExecutionRole'),
      );
```

Reply:
I see. Maybe the maintainer will request you to move the fix to another PR because it is a related but different issue.

Reply:
Don't forget to revert this change :)